src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
> Possible 0-255 values do not seem to stop the current implementation from handling explicit TINYINT metadata mapping it to Java byte type (-128..127):

That logic looks safe to me. `resultSet.getByte()` is JDBC driver-specific behavior for treating the column value as a byte, but the expectation is that it does the right thing and returns a signed byte. For example, if you try with SQLite JDBC a select query for "SELECT 255" and `getByte()` on the result correctly gives me `-1`.